pkg/client/consensuscmd.go

```
@@ -67,25 +67,31 @@ func EstimatedHeightAt(t time.Time) types.BlockHeight {
 }
 
 // Consensustransactioncmd is the handler for the command `rivinec consensus transaction`.
-// Prints the transaction found for the given shortID.
-func Consensustransactioncmd(shortID string) {
-	if !consensusShortIDRegexp.MatchString(shortID) {
-		Die("invalid shortID: ", shortID)
+// Prints the transaction found for the given id. If the ID is a long transaction ID, it also
+// prints the short transaction ID for future reference
+func Consensustransactioncmd(id string) {
+	var txn api.ConsensusGetTransaction
+	var err error
+	if consensusLongIDRegexp.MatchString(id) {
+		err = GetAPI("/consensus/transactions/longid/"+id, &txn)
+	} else if consensusShortIDRegexp.MatchString(id) {
+		err = GetAPI("/consensus/transactions/shortid/"+id, &txn)
+	} else {
+		Die("invalid ID: ", id)
 	}
 
-	var txn api.ConsensusGetTransaction
-	err := GetAPI("/consensus/transactions/"+shortID, &txn)
 	if err != nil {
-		Die("failed to get transaction: ", err, "; shortID: ", shortID)
+		Die("failed to get transaction: ", err, "; ID: ", id)
 	}
 
 	encoder := json.NewEncoder(os.Stdout)
 	err = encoder.Encode(txn)
 	if err != nil {
-		Die("failed to encode transaction: ", err, "; shortID: ", shortID)
+		Die("failed to encode transaction: ", err, "; ID: ", id)
 	}
 }
 
 var (
 	consensusShortIDRegexp = regexp.MustCompile("^[0-9]{1,8}$")
+	consensusLongIDRegexp  = regexp.MustCompile("^[a-z0-9]{64}$")
```

If you do really want to keep this Regexp approach it should be `^[a-f0-9]{64}$`, however I start to think it's a bit overkill here. I know I started by adding that first Regexp, but as long as the daemon code is written properly, it should fail anyhow on an invalid string when unmarshalling from it. If so, we can/should keep it as simple as checking for the length.